src/NHibernate/Type/ComponentType.cs

```diff
@@ -520,6 +520,8 @@ public override object SemiResolve(object value, ISessionImplementor session, ob
 		{
 			//note that this implementation is kinda broken
 			//for components with many-to-one associations
```

Comment:
I do not know if that comment should be removed. (Was it there for the trouble showcased in this PR? Who knows/remembers? It was committed in 2008, 2ea2332. @fabiomaulo?)